core/deployment/src/main/java/io/quarkus/deployment/configuration/ConfigDefinition.java

```diff
@@ -264,59 +274,113 @@ private GroupConfigType processConfigGroup(final String containingName, final Co
                 if (rawTypeOfParameter(fieldType, 0) != String.class) {
                     throw reportError(field, "Map key must be " + String.class);
                 }
-                gct.addField(processMap(field.getName(), gct, field, consume, subKey, typeOfParameter(fieldType, 1),
-                        accessorFinder, javadocKey));
+
+                Type mapValueType = typeOfParameter(fieldType, 1);
+                Class<?> mapValueRawType = mapValueType instanceof ParameterizedType ? rawTypeOf(mapValueType)
```

Comment:
You can just make this be `Class<?> mapValueRawType = rawTypeOf(mapValueType);` and it should work fine.